### PR TITLE
DM-39206: Power On ATCalSys SAL script

### PR DESCRIPTION
**Pull request by isotuela**

### CODE REVIEW

File: doc/version_history.rst [hunk 1]
```diff
@@ -29,6 +33,7 @@ v1.24.0
 * Patch ``base_block_script.py`` to add ``test_case`` attribute.
 * Add new ``maintel/m1m3/enable_m1m3_balance_system.py`` and ``maintel/m1m3/disable_m1m3_balance_system.py`` sal scripts and associated files.
 
+
```
Comment (tribeiro) [inline doc/version_history.rst:36]: Can you delete this empty line?

File: doc/version_history.rst [hunk 2]
```diff
@@ -5,6 +5,10 @@
 ===============
 Version History
 ===============
+v1.25.0
```
Comment (tribeiro) [inline doc/version_history.rst:8]: Can you add an empty line before this line?

File: python/lsst/ts/standardscripts/auxtel/calibrations/power_on_atcalsys.py [hunk 3]
```diff
@@ -0,0 +1,299 @@
...
+        await self.checkpoint("Turning on lamp")
+        await self.switch_lamp_on()
+
+        await self.checkpoint("Waiting for lamp to warm up")
+        await self.wait_for_lamp_to_warm_up()
+
+        await self.checkpoint("Configuring ATMonochromator")
+        await self.configure_atmonochromator()
+
+    async def start_chiller(self):
```
Comment (tribeiro) [inline python/lsst/ts/standardscripts/auxtel/calibrations/power_on_atcalsys.py:184]: Please, add docstring to this method.

File: python/lsst/ts/standardscripts/auxtel/calibrations/power_on_atcalsys.py [hunk 4]
```diff
@@ -0,0 +1,299 @@
...
+        await self.white_light_source.cmd_setChillerTemperature.set_start(
+            temperature=self.chiller_temperature, timeout=self.cmd_timeout
+        )
+        await self.white_light_source.cmd_startChiller.start(
+            timeout=self.timeout_chiller_cool_down
+        )
+
+    async def wait_for_chiller_temp_within_tolerance(self):
```
Comment (tribeiro) [inline python/lsst/ts/standardscripts/auxtel/calibrations/power_on_atcalsys.py:192]: Please add docstring.

File: python/lsst/ts/standardscripts/auxtel/calibrations/power_on_atcalsys.py [hunk 5]
```diff
@@ -0,0 +1,299 @@
...
+                / chiller_temps.setTemperature
+                <= self.chiller_temp_tolerance_relative
+            ):
+                chill_time = time.time() - start_chill_time
+                self.log.info(
+                    f"Chiller reached target temperature, {tel_chiller_temp:0.1f} deg"
+                    f"within tolerance in {chill_time:0.1f} s."
+                )
+                break
+            else:
```
Comment (tribeiro) [inline python/lsst/ts/standardscripts/auxtel/calibrations/power_on_atcalsys.py:214]: I think this `else` statement is moot, you can probably remove it (and remove `continue` as well).  Usually we do this if there's anything else that needs to be done in the loop and we want to escape, however this is not the case here as there's nothing else below it.

File: python/lsst/ts/standardscripts/auxtel/calibrations/power_on_atcalsys.py [hunk 6]
```diff
@@ -0,0 +1,299 @@
...
+                continue
+        else:
+            raise TimeoutError(
+                f"Timeout waiting after {self.timeout_chiller_cool_down} s "
+                f"for the chiller to chill to {chiller_temps.setTemperature} deg. "
+                f"Stayed at {tel_chiller_temp:0.1f} deg."
+            )
+
+    async def open_white_light_shutter(self):
```
Comment (tribeiro) [inline python/lsst/ts/standardscripts/auxtel/calibrations/power_on_atcalsys.py:223]: Please, add docstring.

File: python/lsst/ts/standardscripts/auxtel/calibrations/power_on_atcalsys.py [hunk 7]
```diff
@@ -0,0 +1,299 @@
...
+        await self.white_light_source.cmd_openShutter.start(
+            timeout=self.timeout_open_shutter
+        )
+
+    async def switch_lamp_on(self):
```
Comment (tribeiro) [inline python/lsst/ts/standardscripts/auxtel/calibrations/power_on_atcalsys.py:228]: Please add docstring.

File: python/lsst/ts/standardscripts/auxtel/calibrations/power_on_atcalsys.py [hunk 8]
```diff
@@ -0,0 +1,299 @@
...
+        self.white_light_source.evt_lampState.flush()
+
+        await self.white_light_source.cmd_turnLampOn.set_start(
+            power=self.whitelight_power, timeout=self.timeout_lamp_warm_up
+        )
+
+    async def wait_for_lamp_to_warm_up(self):
```
Comment (tribeiro) [inline python/lsst/ts/standardscripts/auxtel/calibrations/power_on_atcalsys.py:235]: Please, add dosctring.

File: python/lsst/ts/standardscripts/auxtel/calibrations/power_on_atcalsys.py [hunk 9]
```diff
@@ -0,0 +1,299 @@
...
+                )
+                self.log.info(
+                    f"Lamp state: {ATWhiteLight.LampBasicState(lamp_state.basicState)!r}."
+                )
+            except asyncio.TimeoutError:
+                raise RuntimeError(
+                    f"White Light Lamp failed to turn on after {self.timeout_lamp_warm_up} s."
+                )
+
+    async def configure_atmonochromator(self):
```
Comment (tribeiro) [inline python/lsst/ts/standardscripts/auxtel/calibrations/power_on_atcalsys.py:258]: Please, add docstring.

File: python/lsst/ts/standardscripts/auxtel/calibrations/power_on_atcalsys.py [hunk 10]
```diff
@@ -0,0 +1,299 @@
...
+        params = await self.get_monochromator_parameters()
+
+        self.log.info(
+            f"ATMonochromator grating is {params[0]}, \n"
+            f"wavelength is {params[1]}, \n"
+            f"with entry slit width {params[2]}, \n"
+            f"and exit slit width {params[3]}"
+        )
+
+    async def get_monochromator_parameters(self):
```
Comment (tribeiro) [inline python/lsst/ts/standardscripts/auxtel/calibrations/power_on_atcalsys.py:284]: Please, add docstring.

File: python/lsst/ts/standardscripts/auxtel/calibrations/power_on_atcalsys.py [hunk 11]
```diff
@@ -0,0 +1,299 @@
...
+        return await asyncio.gather(
+            self.monochromator.evt_selectedGrating.aget(timeout=self.cmd_timeout),
+            self.monochromator.evt_wavelength.aget(timeout=self.cmd_timeout),
+            self.monochromator.evt_entrySlitWidth.aget(timeout=self.cmd_timeout),
+            self.monochromator.evt_exitSlitWidth.aget(timeout=self.cmd_timeout),
+        )
+
+    async def assert_components_enabled(self):
+        """Check if ATWhiteLight and ATMonochromator are ENABLED"""
```
Comment (tribeiro) [inline python/lsst/ts/standardscripts/auxtel/calibrations/power_on_atcalsys.py:293]: Please, add a `Raises` session to the docstring and explain that it raises a `RuntimeError`, e.g.;

```
Raises
------
RuntimeError:
  If ATWhiteLight or ATMonochromator are not in ENABLED state.
```

**Review by tribeiro — COMMENTED** [review]
Alright one more round of comments. Pretty minor comments now, once you take care of these I think I will be ready to approve the PR.

Thanks!

File: doc/version_history.rst (hunk 2 above)
Comment (isotuela) [reply doc/version_history.rst:9]: I think I messed up the spaces with the rebase conflict.

**Review by tribeiro — APPROVED** [review]
Looks good! thanks for patience!

**Pull request merged**